app/api/mongodb/seed_script.js

```diff
@@ -0,0 +1,136 @@
+}
+
+function listFiles(dirPaths) {
+    let objs = {}
+    function readDirSync(currPath) {
+        const entries = fs.readdirSync(currPath, {withFileTypes: true})
+        entries.forEach((entry) => {
+            const fullPath = path.join(currPath, entry.name)
+            if (entry.isDirectory()) {
+                readDirSync(fullPath);
+            } else if (entry.isFile()) {
+                const fileData = JSON.parse(fs.readFileSync(fullPath, "utf8"));
+                let parseParentPath = entry.parentPath.split("/");
+                let book = parseParentPath.pop();
+                book = booksMapping[book];
+                let chapters = fileData.text;
+                let translation = fileData.version; // ONLY DO ESV, NASB, and KJV
+                if (translation === "ESV" || translation === "NASB" || translation === "KJV") {
+                    if (!objs[book]) objs[book] = {
+                        id: book,
+                        chapters: Array(chapters.length)
+                    }
+                    for (let i = 0; i < chapters.length; i++) {
+                        let chapter = chapters[i];
+                        let verses = chapter.text;
+                        let numVerses = Number(verses[verses.length - 1]["ID"])
+                        if (!objs[book].chapters[i]) {
+                            let chObjs = {
+                                id: book + "_" + String(i+1),
+                                verses: Array(numVerses)
+                            }
+                            objs[book].chapters[i] = chObjs
+                        }
+                        for (let j = 0; j < verses.length; j++) {
+                            let verse = verses[j];
+                            let verseId = Number(verse["ID"]) - 1 // turn ID to number for index
+                            let verseText = verse["text"]
+                            if (!objs[book].chapters[i].verses[verseId]) {
+                                objs[book].chapters[i].verses[verseId] = {
+                                    id: book + "_" + String(i+1) + "_" + verse["ID"]
+                                }
+                            }
+                            if (!(objs[book].chapters[i].verses[verseId][translation])) {
+                                objs[book].chapters[i].verses[verseId] = {
+                                    ...objs[book].chapters[i].verses[verseId],
+                                    [translation]: verseText
+                                }
+                            } else {
+                                objs[book].chapters[i].verses[verseId][translation] += verseText ? " " + verseText : ""
+                            }
+                        }
+                    }
+                }
+            }
+        })
+
+    }
+    dirPaths.forEach((path) => {
+        readDirSync(path)
+
+    })
+    fs.writeFileSync("data.json", JSON.stringify(objs, null, 4), "utf8");
```

Comment:
I think writing the file here is ok for the sake of verifying the results, but does this script update the mongo db anywhere?

Also, we shouldn't check in the resulting data.json. That should either be in the .gitignore or deleted

Reply:
created .gitignore to hide data.json